staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -85,6 +86,9 @@ type Config struct {
 	// needs to process an incoming event.
 	TriggerPublisherFunc storage.TriggerPublisherFunc
 
+	// NewFunc is a function that creates new empty object storing a object of type Type.
+	NewFunc func() runtime.Object
```

Comment:
Can you please change that so that:
- you only pass NewFunc via config
- you remove Type field from config (this is initialited to NewFunc() anyway)

- move that change to a separate PR - this is non-controversial and we can merge it soon.